src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompareTests.cs

```diff
@@ -0,0 +1,277 @@
+
+        public static IEnumerable<object[]> CS_IneligibleStringCompareOverloadTestData
+        {
+            get
+            {
+                yield return new[] { "string.Compare(x, y, true, System.Globalization.CultureInfo.InvariantCulture)" };
+                yield return new[] { "string.Compare(x, y, System.Globalization.CultureInfo.InvariantCulture, System.Globalization.CompareOptions.None)" };
+            }
+        }
+
+        public static IEnumerable<object[]> VB_IneligibleStringCompareOverloadTestData
+        {
+            get
+            {
+                yield return new[] { "String.Compare(x, y, true, System.Globalization.CultureInfo.InvariantCulture)" };
+                yield return new[] { "String.Compare(x, y, System.Globalization.CultureInfo.InvariantCulture, System.Globalization.CompareOptions.None)" };
+            }
+        }
+
+        #endregion
+
+        [Theory]
+        [MemberData(nameof(CS_ComparisonLeftOfLiteralTestData))]
+        [MemberData(nameof(CS_ComparisonRightOfLiteralTestData))]
+        [MemberData(nameof(CS_InvertedComparisonLeftOfLiteralTestData))]
+        [MemberData(nameof(CS_InvertedComparisonRightOfLiteralTestData))]
+        public Task StringCompareResult_CompareToZero_Diagnostic_CS(string testExpression, string fixedExpression)
+        {
+            string testCode = $@"
+using System;
+
+public class Testopolis
+{{
+    public bool Huh(string x, string y)
+    {{
+        return {{|#0:{testExpression}|}};
+    }}
+}}";
+            string fixedCode = $@"
+using System;
+
+public class Testopolis
+{{
+    public bool Huh(string x, string y)
+    {{
+        return {fixedExpression};
+    }}
+}}";
+
+            return VerifyCS.VerifyCodeFixAsync(testCode, VerifyCS.Diagnostic(Rule).WithLocation(0), fixedCode);
+        }
+
+        [Theory]
+        [MemberData(nameof(VB_ComparisonLeftOfLiteralTestData))]
+        [MemberData(nameof(VB_ComparisonRightOfLiteralTestData))]
+        [MemberData(nameof(VB_InvertedComparisonLeftOfLiteralTestData))]
+        [MemberData(nameof(VB_InvertedComparisonRightOfLiteralTestData))]
+        public Task StringCompareResult_CompareToZero_Diagnostic_VB(string testExpression, string fixedExpression)
+        {
+            string testCode = $@"
+Imports System
+
+Public Class Testopolis
+
+    Public Function Huh(x As String, y As String) As Boolean
+        Return {{|#0:{testExpression}|}}
+    End Function
+End Class";
+            string fixedCode = $@"
+Imports System
+
+Public Class Testopolis
+
+    Public Function Huh(x As String, y As String) As Boolean
+        Return {fixedExpression}
+    End Function
+End Class";
+
+            return VerifyVB.VerifyCodeFixAsync(testCode, VerifyVB.Diagnostic(Rule).WithLocation(0), fixedCode);
+        }
+
+        [Theory]
+        [MemberData(nameof(CS_StringCompareExpressionsTestData))]
+        public Task StringCompareResult_CompareToNonLiteralZero_NoDiagnostic_CS(string expression)
```

Comment:
For const values we could also provide a diagnostic. I'd consider this an area for future improvement and not required for this PR

Reply:
If I write an expression with a constant in it, I expect that changing the value initially assigned to the constant will change the value of the expression. The fixer removes the constant from the expression entirely, which could break if the constant is given a different value.

Reply:
While true, it's an indicator of what I would consider hard to read code and potentially bad design. I don't know of a strong use case where we would want to encourage that, and analyzers are opinionated by design.